server/cp/macros/update_translation_metadata_macro.py

```diff
@@ -26,21 +34,37 @@ def update_translation_metadata_macro(item, **kwargs):
     if not cv or not cv.get("items"):
         return
 
-    subject = item.get("subject", [])
-    is_destination = any(
-        sub for sub in subject if sub.get("name") == "Presse Canadienne staff"
-    )
-
-    for value in cv["items"]:
-        if value.get("name") == "Presse Canadienne staff" and not is_destination:
-            item.setdefault("subject", []).append(
-                {
-                    "name": value["name"],
-                    "qcode": value["qcode"],
-                    "scheme": "destinations",
-                }
-            )
-            break
+    subjects = item.get("subject", [])
+
+    # Remove out all the fields which are of no use.
+    subjects = [sub for sub in subjects if sub.get("name") in fields and sub.get("scheme") == "destinations"]
```

Comment:
To follow on above comment, it would be better to use `qcode` instead of `name` here.